### PR TITLE
feat(models): enable hybrid forces for Ewald and PME wrappers

### PR DESCRIPTION
**Pull request by zubatyuk**
# ALCHEMI Toolkit Pull Request

## Description

Enable hybrid force computation for Ewald and PME electrostatic wrappers. With `hybrid_forces=True`, the nvalchemiops kernels return analytical direct forces (`dE/dR|_q`) detached from the autograd graph while injecting `dE/dq` into the energy via `_InjectChargeGrad`. The pipeline then adds the charge chain-rule contribution `(dE/dq)(dq/dR)` via autograd on the summed energy, giving the correct total force for models with geometry-dependent charges `q(R)`.

## Type of Change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Performance improvement
- [ ] Documentation update
- [ ] Refactoring (no functional changes)
- [ ] CI/CD or infrastructure change

## Related Issues

Requires nvalchemiops >= 0.3.1.

## Changes Made

- Pass `hybrid_forces=True` to `ewald_real_space`, `ewald_reciprocal_space`, and `particle_mesh_ewald` kernel calls in `EwaldModelWrapper` and `PMEModelWrapper`.
- Set `autograd_outputs=frozenset({"forces"})` in both wrappers so the pipeline knows to compute autograd derivatives from the energy and sum them with direct kernel forces.
- Fix `PipelineModelWrapper._run_autograd_group` to sum direct additive outputs (e.g. detached kernel forces) from step outputs alongside autograd-derived derivatives, preventing silent drop of direct forces.
- Update module-level and class docstrings in `ewald.py` and `pme.py` to describe hybrid force semantics and the charge gradient pathway.

## Testing

- [x] Unit tests pass locally (`make pytest`)
- [x] Linting passes (`make lint`)
- [x] New tests added for new functionality meets coverage expectations?

New tests added:
- `test/models/test_ewald.py`: `TestEwaldHybridForces` class with 7 tests covering autograd_outputs, energy/forces shape, grad_fn presence/absence, charge gradient via backward, and numerical comparison against non-hybrid path.
- `test/models/test_pme.py`: 6 hybrid force tests covering the same aspects for PME.
- `test/models/test_pipeline.py`: `TestAutoGradGroupHybridForces` class with 3 tests verifying that direct forces are correctly summed with autograd forces in the pipeline, energy-only models work, and hybrid forces exceed autograd-only forces.

## Checklist

- [x] I have read and understand the [Contributing Guidelines](../CONTRIBUTING.md)
- [ ] I have updated the [CHANGELOG.md](../CHANGELOG.md)
- [x] I have performed a self-review of my code
- [x] I have added docstrings to new functions/classes
- [x] I have updated the documentation (if applicable)

## Additional Notes

This change is backward-compatible: `hybrid_forces=True` is always passed unconditionally. When `charges.requires_grad=False` (fixed charges), the kernel returns identical forces and a plain energy tensor with no grad_fn — behavior is unchanged from the non-hybrid path. The charge gradient pathway only activates when `charges.requires_grad=True`.

> [!TIP]
> This repository uses Greptile, an AI code review service, to help conduct
> pull request reviews. We encourage contributors to read and consider suggestions
> made by Greptile, but note that human maintainers will provide the necessary
> reviews for merging: Greptile's comments are **not** a qualitative judgement
> of your code, nor is it an indication that the PR will be accepted/rejected.
> We encourage the use of emoji reactions to Greptile comments, depending on
> their usefulness and accuracy.

### CODE REVIEW

File: nvalchemi/models/pipeline.py [hunk 1]
```diff
@@ -693,6 +693,17 @@ def _run_autograd_group(
...
+        # Sum direct additive outputs from step outputs (e.g. hybrid-force
+        # models that return detached kernel forces) alongside the autograd
+        # derivatives computed above.
+        for o in step_outputs:
+            for key, val in o.items():
+                if val is not None and key in self.additive_keys and key != "energy":
+                    if key in group_out and group_out[key] is not None:
+                        group_out[key] = group_out[key] + val
+                    else:
+                        group_out[key] = val
```
[the remark this answers is not in the document]
Comment (zubatyuk) [reply nvalchemi/models/pipeline.py:708]: Addressed in c387d8b09f670db7ca888ae6396fad9075a9ebb4.

File: test/models/test_ewald.py [hunk 2]
```diff
@@ -589,3 +623,221 @@ def test_neutral_system_forces_symmetry(self):
...
+
+    @pytest.fixture(autouse=True)
+    def _require_ops(self):
+        pytest.importorskip("nvalchemiops")
+
+    @staticmethod
+    def _build_nl(batch, model):
+        from nvalchemi.neighbors import compute_neighbors
+
+        compute_neighbors(batch, config=model.model_config.neighbor_config)
```
[the remark this answers is not in the document]
Comment (zubatyuk) [reply test/models/test_ewald.py:644]: Fixed in 8e4dd9e5c4203358bf8c9a79fdf09ec652e42825.

File: test/models/test_pme.py [hunk 3]
```diff
@@ -605,6 +639,181 @@ def test_ewald_and_pme_agree_on_energy_sign(self):
...
+            mask_value=fill_value,
+            compute_forces=False,
+            compute_virial=True,
+            accuracy=w.accuracy,
+            hybrid_forces=False,
+        )
+        expected_stress = result[1] * w.coulomb_constant
+
+        torch.testing.assert_close(
+            out_hybrid["stress"], expected_stress, atol=1e-5, rtol=1e-5
```
[the remark this answers is not in the document]
Comment (zubatyuk) [reply test/models/test_pme.py]: Fixed in d820d13aaf3b6560c24e4a2f4c65b023163d1d8d.
